### PR TITLE
#DP-7 feat: admin trainee dashboard

### PR DESCRIPTION
**Pull request by chanceDira**
## PR Description
Admin trainee dashboard 

## Testing the repo
```
git clone https://github.com/atlp-rwanda/atlp-pulse-fn.git
cd atlp-pulse-fn
git checkout ft-admin-trainee-dashboard-dp-7
npm i
npm run dev
```
direct to 
http://localhost:3000/dashboard/trainees
http://localhost:3000/dashboard/tratings
http://localhost:3000/dashboard/updated-ratings

## Checklist 
- [x] Available trainees
- [x] User ratings
- [x] Filter cohort, ratings, and program
- [x] Search by name
- [x] Trainee rating
- [x] Accept/reject rating updates

## Screenshot

Screen Shot 2022-07-14 at 21 24 31

Screen Shot 2022-07-14 at 21 24 49

Screen Shot 2022-07-14 at 21 24 55

Screen Shot 2022-07-14 at 21 25 08

Screen Shot 2022-07-22 at 10 55 14

(Need reviews)
[#DP-7](https://trello.com/c/s6IiesL2/7-trainee-dashboard-admin)

### CODE REVIEW

**Comment by feyton** [comment]
I deployed a review app for you to try out this pull request 👇🏽

Heroku App: https://devpulse-review-app-26.herokuapp.com/


<hidden purpose="for-rewritable-pr-comment-action-use" value="Deploy"></hidden>

**Review by samyshyaka — DISMISSED** [review]
Can you find the reason why the review app failed to deploy? Currently can't test your work using the review app

**Review by jeandamascyifuzo — APPROVED** [review]

File: src/components/Sidebar.tsx [hunk 1]
```diff
@@ -14,71 +14,69 @@ import {
 } from '@heroicons/react/solid';
 import { CogIcon } from '@heroicons/react/outline';
 
-const Sidebar = ({ style }: { style: string }) => {
+function Sidebar({ style }: { style: string }) {
   const { t } = useTranslation();
   return (
     <div
-      className={`${style} flex-col fixed h-[100%] lg:w-[30vh] pt-[3vh] lg:pt-[11vh] bg-white dark:bg-dark-bg border-r py-4 pl-4`}
+      className={`${style} flex-col fixed h-[100%] lg:w-[20vh] pt-[3vh] lg:pt-[11vh] bg-white dark:bg-dark-bg border-r py-4 pl-4`}
```
Comment (feyton) [inline src/components/Sidebar.tsx:21]: We have two sidebars which one are you using in your project?

File: src/components/Siderbar.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+import {
```
Comment (feyton) [inline src/components/Siderbar.tsx:3]: Can you replace this file with the official sidebar which have links and translated texts

File: src/pages/AdminTraineeDashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+const AdminTraineeDashboard = () => {
+  const [registerTraineeModel, setRegisterTraineeModel] = useState(false);
+
+  const removeModel = () => {
+    let newState = !registerTraineeModel;
+    setRegisterTraineeModel(newState);
+  };
+
+  let developers = [
```
Comment (feyton) [inline src/pages/AdminTraineeDashboard.tsx:13]: move these data into a file and import them from there.

We will replace these after adding a json server

**Review by feyton — CHANGES_REQUESTED** [review]
### Changes
@chanceDira Please refer to the comments especially regarding removing data from the `tsx` files and move them outside. Since we will be adding the Json Server

File: src/components/Siderbar.tsx (hunk 2 above)
Comment (chanceDira) [reply src/components/Siderbar.tsx:3]: ok, let do that.

File: src/pages/AdminTraineeDashboard.tsx (hunk 3 above)
Comment (chanceDira) [reply src/pages/AdminTraineeDashboard.tsx:13]: solved

File: src/components/Siderbar.tsx (hunk 2 above)
Comment (chanceDira) [reply src/components/Siderbar.tsx:3]: solved

**Review by feyton — APPROVED** [review]

**Review by ceelogre — COMMENTED** [review]
Please raise coverage.

**Review by samyshyaka — CHANGES_REQUESTED** [review]
Great work @chanceDira,

- Something is wrong with the width of the sidebar and is causing the icons to not be visible. Check if you rebased correctly
- The  background color of the input fields on registering a new trainee in the light mode isn't right
- On the drop-downs of the 'cohorts' and 'phases' the padding on the x-axis is too big. Try to reduce it.
- Also, try to find a way to make the drop-downs disappear once the user clicks outside

**Review by samyshyaka — APPROVED** [review]

**Comment by Julesbi** [comment]
Nice Jobc@chanceDira though I have one change to request, the size of dropdowns are too small on the page of rating trainees could increase them and try to center them.

**Review by Julesbi — CHANGES_REQUESTED** [review]
Nice Job@chanceDira though I have one change to request, the size of dropdowns are too small on the page of rating trainees could increase them and try to center them.

**Comment by feyton** [comment]
> 1. urls like `dashboard/trainee-rating-dashboard` are not consistent with the standard naming convention (for starters, the word dashboard is repeated in the same url) so they should be changed and shortened.
> 
>     2. Some of the checklist items were not met, I was unable to see filter by cohort and name.
> 
>     3. On the trainee rating dashboard, why is there an edit action available? Ideally, editing is for coordinators so admins should only be able to view.
> 
>     4. On the updated rating dashboard, under what circumstances can there be a change in sprint?
> 
>     5. Finally, If I login as Admin, where should I click to access all of the three interfaces?

Good morning, even though I have not worked on this feature. 
1. Some requested changes are heavily reliant on #PR-30 which I am working on. I am aware about the coverage issue and I have no means of raising the coverage to the desired percentage but the features are working. If that would be possible, merging this PR will be useful and will allow other team members to test different scenarios before shipping the product with using many components where the difference maybe a `title` for different roles.

2. Also, can you provide more clarity on point 5 @ceelogre . It looks like the feature am working on where a user is only meant to view his own aspect of the application without the possibility to see the other parts despite the role. 
> Scenario, the admin should not be able to see the trainee performance page since it tailored for a single trainee. He will have access to the page where he will be able to rate the trainee instead.

**Review by ceelogre — COMMENTED** [review]
on updated ratings, keep the sprint column so the admin which sprint an updated rating belongs to.

**Review by Julesbi — APPROVED** [review]

**Comment by chanceDira** [comment]
> 1. urls like `dashboard/trainee-rating-dashboard` are not consistent with the standard naming convention (for starters, the word dashboard is repeated in the same url) so they should be changed and shortened.
> 2. Some of the checklist items were not met, I was unable to see filter by cohort and name.
> 3. On the trainee rating dashboard, why is there an edit action available? Ideally, editing is for coordinators so admins should only be able to view.
> 4. On the updated rating dashboard, under what circumstances can there be a change in sprint?
> 5. Finally, If I login as Admin, where should I click to access all of the three interfaces?

**Review by ceelogre — APPROVED** [review]

**Pull request merged**